### PR TITLE
Login Page

### PR DESCRIPTION
**Pull request by Andrew-Kiev**

### CODE REVIEW

File: AutWebTest/AutWebTest/Helpers/Helper.cs [hunk 1]
```diff
@@ -0,0 +1,17 @@
+﻿using AutWebTest.Pages;
+using FizzWare.NBuilder;
+using OpenQA.Selenium;
+using OpenQA.Selenium.Chrome;
+
+namespace AutWebTest.Helpers
+{
+    public static class Helper
+    {
+        public static string RandomString(int length)
```
Comment (varajan) [inline AutWebTest/AutWebTest/Helpers/Helper.cs]: Methods should have some verb in title, e.g. GetRandomString, GenerateRandomString, etc.

```suggestion
        public static string GetRandomString(int length)
```

File: AutWebTest/AutWebTest/Pages/LoginPage.cs [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+        private IWebElement PasswordField => driver.FindElement(By.Id("password"));
+        private IWebElement LoginButton => driver.FindElement(By.Id("loginBtn"));
+        public string ErrorMessage => driver.FindElement(By.Id("errorMessage")).Text;
+
+        public void Login(string login, string password)
+        {
+            //if (LoginField.GetProperty. != string.Empty)
+            //    LoginField.Clear();
+            //if (PasswordField.GetProperty. != string.Empty)
+            //    PasswordField.Clear();
```
Comment (varajan) [inline AutWebTest/AutWebTest/Pages/LoginPage.cs]: Please delete commented code before merge.

File: AutWebTest/AutWebTest/Tests/LoginPageTests.cs [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+        public void LoginWithEmptyField()
+        {
+            loginPage.Login(string.Empty, string.Empty);
+            Assert.AreEqual("User not found!", loginPage.ErrorMessage);
+        }
+
+        [Test]
+        public void LoginWithEmptyName()
+        {
+
```
Comment (varajan) [inline AutWebTest/AutWebTest/Tests/LoginPageTests.cs]: Blank line

File: AutWebTest/AutWebTest/Tests/LoginPageTests.cs [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+        [TearDown]
+        public void TearDown() => driver.Dispose();
+
+        [Test]
+        public void LoginWithEmptyField()
+        {
+            loginPage.Login(string.Empty, string.Empty);
+            Assert.AreEqual("User not found!", loginPage.ErrorMessage);
+        }
```
Comment (varajan) [inline AutWebTest/AutWebTest/Tests/LoginPageTests.cs]: You have lots of similar tests. You can combine them like this:
```suggestion
        [TestCase(string.Empty, string.Empty, "User not found!")]
        [TestCase(string.Empty, "newyork1", "User not found!")]
        [TestCase("test", string.Empty, "User not found!")]
        public void NegativeLogin(string login, string password, string expectedError)
        {
            loginPage.Login(login, password);
            Assert.AreEqual(expectedError, loginPage.ErrorMessage);
        }
```

**Review by varajan — COMMENTED** [review]
Nice and good. Have few code improvements.

File: AutWebTest/AutWebTest/Pages/LoginPage.cs [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+            LoginField.SendKeys(login);
+            PasswordField.SendKeys(password);
+            LoginButton.Click();
```
Comment (varajan) [inline AutWebTest/AutWebTest/Pages/LoginPage.cs:33]: Add an explicit wait here. You should wait for an error is shown or page url is changed.

File: AutWebTest/AutWebTest/Tests/LoginPageTests.cs (hunk 4 above)
Comment (Andrew-Kiev) [reply AutWebTest/AutWebTest/Tests/LoginPageTests.cs]: Good suggestion, thank you very much!

**Review by varajan — DISMISSED** [review]

**Review by varajan — APPROVED** [review]

**Pull request merged**